src/PrestaShopBundle/Command/ShopCreatorCommand.php

```diff
@@ -0,0 +1,247 @@
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Command;
+
+use Address;
+use Cart;
+use CartRule;
+use Customer;
+use Db;
+use Faker\Factory;
+use Order;
+use OrderState;
+use Product;
+use Shop;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * This command is used for appending the hook names in the configuration file.
+ */
+class ShopCreatorCommand extends Command
+{
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    public function __construct(string $dbPrefix)
+    {
+        parent::__construct();
+        $this->dbPrefix = $dbPrefix;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('prestashop:shop-creator')
+            ->addOption('orders', null, InputOption::VALUE_OPTIONAL, 'Number of orders to create', 10000)
+            ->addOption('customers', null, InputOption::VALUE_OPTIONAL, 'Number of customers without order to create', 5000)
+            ->addOption('carts', null, InputOption::VALUE_OPTIONAL, 'Number of carts to create', 50000)
+            ->addOption('cart-rules', null, InputOption::VALUE_OPTIONAL, 'Number of cart rules to create', 10000)
+            ->addOption('products', null, InputOption::VALUE_OPTIONAL, 'Number of products to create', 10000)
+            ->addOption('shopId', null, InputOption::VALUE_OPTIONAL, 'The shop identifier', 1)
+            ->addOption('shopGroupId', null, InputOption::VALUE_OPTIONAL, 'The shop group identifier', 1)
+            ->addOption('languageId', null, InputOption::VALUE_OPTIONAL, 'The languageId identifier', 1)
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        \Context::getContext()->currency = \Currency::getDefaultCurrency();
+
+        $faker = Factory::create('fr_FR');
+
+        $numberOfOrders = (int) $input->getOption('orders');
+        $numberOfCustomerWithoutOrder = (int) $input->getOption('customers');
+        $numberOfCarts = (int) $input->getOption('carts');
+        $numberOfCartRules = (int) $input->getOption('cart-rules');
+        $numberOfProducts = (int) $input->getOption('products');
+        $idLang = (int) $input->getOption('languageId');
+        $idshop = (int) $input->getOption('shopId');
+        $idShopGroup = (int) $input->getOption('shopGroupId');
+
+        $productIds = $this->getSimpleProducts($idLang);
```

Comment:
If you know that you generate X products, I think you could get random numbers between starting id and max id instead of querying products